File: sdk/python/cmd/pulumi-language-python/testdata/setuppy/published/projects/l3-component-simple/myComponent.py
import pulumi
from pulumi import Input
from typing import Optional, Dict, TypedDict, Any
import pulumi_simple as simple

class MyComponentArgs(TypedDict, total=False):
    input: Input[bool]

class MyComponent(pulumi.ComponentResource):
    def __init__(self, name: str, args: MyComponentArgs, opts:Optional[pulumi.ResourceOptions] = None):
        super().__init__("components:index:MyComponent", name, args, opts)

        res = simple.Resource(f"{name}-res", value=args["input"],
        opts = pulumi.ResourceOptions(parent=self))

        self.output = res.value
        self.register_outputs({
            'output': res.value
        })

File: sdk/nodejs/cmd/pulumi-language-nodejs/testdata/published/bun/projects/l3-component-simple/myComponent.ts
import * as pulumi from "@pulumi/pulumi";
import * as simple from "@pulumi/simple";

interface MyComponentArgs {
    /**
     * A simple input
     */
    input: pulumi.Input<boolean>,
}

export class MyComponent extends pulumi.ComponentResource {
    public output: pulumi.Output<boolean>;
    constructor(name: string, args: MyComponentArgs, opts?: pulumi.ComponentResourceOptions) {
        super("components:index:MyComponent", name, args, opts);
        const res = new simple.Resource(`${name}-res`, {value: args.input}, {
            parent: this,
        });

        this.output = res.value;
        this.registerOutputs({
            output: res.value,
        });
    }
}
